src/Runtime/Distributions/Automata/Automaton.Simplification.cs

```diff
@@ -673,7 +673,7 @@ public override string ToString()
                 {
                     var sequence = new Simplification.GeneralizedSequence(currentSequenceElements);
                         // TODO: use immutable data structure instead of copying sequences
-                    weightedSequences.Add(Pair.Create(sequence, Weight.Product(currentWeight, state.EndWeight)));
+                    weightedSequences.Add(ValueTuple.Create(sequence, Weight.Product(currentWeight, state.EndWeight)));
```

Comment:
Instead of ValueTuple, this code should be using a custom struct WeightedSequence with fields "Sequence" and "Weight".